src/Sarif/Readers/DeferredDictionary.cs

```diff
@@ -0,0 +1,514 @@
+
+        /// <summary>
+        ///  JsonObjectMemberStreamReader is used to get JsonTextReader to read a property name and
+        ///  value somewhere in an outer object.
+        ///  
+        ///  It alters the first read so that there's a fake outer object (a starting '{') and
+        ///  whitespace and any comma before the property name are hidden.
+        /// </summary>
+        private class JsonObjectMemberStreamReader : StreamReader
+        {
+            private bool _readCalled;
+
+            public JsonObjectMemberStreamReader(Stream stream) : base(stream, Encoding.Default, true, 1024, true)
+            { }
+
+            public override int Read(char[] buffer, int index, int count)
+            {
+                int countRead = 0;
+
+                // We are seeking to a position just before a new PropertyName in an object.
+                if(!_readCalled)
+                {
+                    _readCalled = true;
+
+                    // We'll add a '{' to the JsonTextReader understands we're in an object
+                    buffer[index++] = '{';
+
+                    // Read the remaining desired count
+                    countRead = base.Read(buffer, index + 1, count - 1);
+
+                    // Make everything before the next PropertyName whitespace (old space and comma between items)
+                    for(int i = index + 1; i < index + 1 + countRead; ++i)
+                    {
+                        if (buffer[i] == '"') break;
+                        buffer[i] = ' ';
+                    }
+
+                    // Return the length found plus our added '{'
+                    return countRead + 1;
+                }
+
+                return base.Read(buffer, index, count);
+            }
+        }
+
+        /// <summary>
+        ///  JsonDeferredDictionaryEnumerator provides enumeration over a Dictionary in a
+        ///  Json object starting at a given position.
+        ///  
+        ///  Items are not pre-loaded and are not kept after enumeration, allowing use with
+        ///  collections too large for memory.
+        /// </summary>
+        /// <typeparam name="U">Dictionary Value item type</typeparam>
+        private class JsonDeferredDictionaryEnumerator<U> : IEnumerator<KeyValuePair<string, U>>
+        {
+            private JsonSerializer _jsonSerializer;
+            private Func<Stream> _streamProvider;
+            private long _start;
+
+            private JsonTextReader _jsonTextReader;
+            private Stream _stream;
+
+            public JsonDeferredDictionaryEnumerator(JsonSerializer jsonSerializer, Func<Stream> streamProvider, long start)
+            {
+                _jsonSerializer = jsonSerializer;
+                _streamProvider = streamProvider;
+                _start = start;
+
+                Reset();
+            }
+
+            public KeyValuePair<string, U> Current { get; private set; }
+            object IEnumerator.Current => Current;
```

Comment:
>object [](start = 12, length = 6)

add 'private' modifier here

Reply:
Hmm. It doesn't compile if I add it. I guess the two 'Currents' conflict between IEnumerable<T> and IEnumerable, so the untyped one has to be implemented explicitly and that's what's happened. This is the code VS adds when you run the 'Implement Interface' code fixer.